### ISSUE TITLE
Parser: cannot combine component with fenced code block

### ISSUE THREAD

**Issue by agentydragon** [athensresearch/athens#1059]
**Problem**

This block:

```
{{[[TODO]]}} 
\`\`\`
hello
\`\`\`
```

Renders as:

![image](https://user-images.githubusercontent.com/714892/116321358-2a473f00-a7ba-11eb-9c7f-e1c4a6236597.png)

I'd expect it to render a pre-formatted code block after the TODO component, instead of parsing the triple-backticks as inline fenced code block containing two backticks on each side.

**Athens Version**

1.0.0-beta.77 454682547046109d0cea3512d17d4fb3095397ec

cc @neotyk

**Comment by agentydragon** [comment]
Another example (remove backslashes before backticks):

```
{{[[TODO]]}} Pandas autocorrelation plot:
\`\`\`
from pandas.plotting import autocorrelation_plot

autocorrelation_plot(series)\`\`\` [[Anki]]
```

Renders as:

![image](https://user-images.githubusercontent.com/714892/116334097-f1689380-a7d4-11eb-8109-cf547e8b41e0.png)

**Comment by neotyk** [comment]
This is because triple backtick is a block element, can you do the same with single backticks (as these are inline not a block level)

**Comment by agentydragon** [comment]
Yeah I figured that's probably the reason. If we want to just not support this, that's also an okay decision, but i think it's still a reasonable thing to want to do as a user. At least, if Athens allowed mixing inline/block elements in a block, it means there's more freedom for the user, and one less rule for the user to make (is, the rule that "components and fenced blocks are block elements and you can't have both in one block")

**Comment by agentydragon** [comment]
Also, would this work if I added newlines around the fenced blocks? I believe that would make it valid Markdown.

**Comment by agentydragon** [comment]
https://github.com/athensresearch/athens/pull/1174 confirms this has indeed been fixed. Thanks!

**Comment by agentydragon** [comment]
Actually turns out this is not fixed yet. @tangjeff0 can you please reopen?

I tried adding a test into `compatibility_test.clj`:

```
diff --git a/test/athens/parser/compatibility_test.clj b/test/athens/parser/compatibility_test.clj
index 97c75a45e..b13bf346c 100644
--- a/test/athens/parser/compatibility_test.clj
+++ b/test/athens/parser/compatibility_test.clj
@@ -95,7 +95,14 @@
     "{{((block-ref-id))}} amazing block"
 
     [:block [:paragraph [:component "AnotherComponent" "AnotherComponent"] " Another Content"]]
-    "{{AnotherComponent}} Another Content"))
+    "{{AnotherComponent}} Another Content"
+
+    ;; Regression test for https://github.com/athensresearch/athens/issues/1059
+    [:block
+     [:paragraph [:component "[[TODO]]" [:page-link "TODO"]]]
+     [:fenced-code-block {:lang ""} [:code-text "some code"]]
+     [:paragraph [:text-run "and some text"]]]
+    "{{[[TODO]]}} ```\nsome code```and some text"))
```

The actual result (after dd5affb08) is:

```
[:block [:paragraph [:component "[[TODO]]" [:page-link "TODO"]] " ``" [:inline-pre-formatted "\nsome code"] "`" "`" "and some text"]]
```

Which is still incorrect.

**Comment by agentydragon** [comment]
Test added in: https://github.com/agentydragon/athens/tree/add-tests

For the failed test run, see: https://github.com/athensresearch/athens/pull/1174/checks?check_run_id=2596303240
